### PR TITLE
Update alpha.kubernetes.io/provided-node-ip

### PR DESCRIPTION
**Pull request by nckturner**
When https://github.com/kubernetes/kubernetes/pull/109794 merges, the following will be true: 

* `alpha.kubernetes.io/provided-node-ip` is no longer set ONLY when --cloud-provider=external
* Now, it is set on kubelet startup if the --cloud-provider flag
  is set at all, including the deprecated in-tree values
  like `aws` and `gcp`.

### CODE REVIEW

**Comment by natalisucks** [comment]
/hold
Until the aforementioned PR is merged, this change should be held. Am happy to approve once that takes place, thanks!

Thanks @nckturner and @dchen1107 for the update and review so far.

**Comment by sftim** [comment]
Should this PR target _dev-1.25_?

**Comment by natalisucks** [comment]
@sftim Given the linked PR (now merged) didn't target dev-1.25, my understanding is that this docs change wouldn't, either. Please feel free to correct me if wrong!

**Comment by nckturner** [comment]
@natalisucks it should target 1.25, is there anything we need to change for the linked PR?  Its got 1.25 as the milestone.

**Comment by sftim** [comment]
The target branch for this PR should be _dev-1.25_ (not _main_). See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request for docs on this if not sure.

**Comment by sftim** [comment]
/lgtm cancel
This will need a fresh LGTM once it's rebased or otherwise updated to target the v1.25 docs branch.

**Comment by jihoon-seo** [comment]
/unhold
since the PR https://github.com/kubernetes/kubernetes/pull/109794 has been merged.

/remove-area blog release-eng
/remove-language ja ko zh
/remove-sig release

**Comment by nckturner** [comment]
Is there anything else to do for this?

**Review by endocrimes — APPROVED** [review]
/lgtm

(node review)

**Pull request merged**
